src/prof_data.c

```diff
@@ -406,7 +406,7 @@ prof_lookup(tsd_t *tsd, prof_bt_t *bt) {
 
 prof_tctx_t *
 prof_tctx_create(tsd_t *tsd) {
-	if (tsd_reentrancy_level_get(tsd) > 0) {
+	if (!tsd_nominal(tsd) || tsd_reentrancy_level_get(tsd) > 0) {
```

Comment:
Does the non nominal check covers the reentrancy one?

Reply:
I think checking `tsd_state_get(tsd) == tsd_state_nominal` would guarantee `tsd_reentrancy_level_get(tsd) == 0`, but `tsd_nominal(tsd)` which checks `tsd_state_get(tsd) <= tsd_state_nominal_max` may not?

Upon a second thought, I think checking `tsd_state_get(tsd) == tsd_state_nominal` feels safer, and it can be checked by `!tsd_fast(tsd)`.

Thoughts?

Reply:
We should still allow the creation during `tsd_state_nominal_slow`; say if we added alloc hooks which switch to slow path, sampling should still be allowed. `tsd_nominal(tsd)` sounds right to me.

Reply:
So let's get rid of the reentrancy condition?

Reply:
Can `tsd_nominal(tsd)` really guarantee `tsd_reentrancy_level_get(tsd) == 0`? I could not find any proof for either case...

Reply:
You are right. No guarantee. Strict nominal isn’t right, e.g. if tcache is off, it’s nominal_slow which should still have profiling working.

Let’s keep it as is then.